### DISCUSSION TITLE
`deno run --vendor --no-remote` does not work

### DISCUSSION THREAD

**Discussion by adamgreg** [denoland/deno#22191] [Q&A] [answered]
Version: Deno 1.40.2

Should the `--no-remote` argument allow imports from a `vendor` folder?

I develop a containerised application that must run without Internet access. I've been trying to switch from using `deno vendor` to using `deno cache --vendor`, followed by `deno run --vendor` to pick up the previously-vendored dependencies.

It seems to work ok (for https deps), but if I try to use the `--no-remote` argument to `deno run` I get ` A remote specifier was requested...` errors.

**Comment by lucacasonato** [comment, marked as answer]
I think you want `--cached-only` rather than `--no-remote`?

**Reply by adamgreg** [reply]
This style of vendoring also seems to require the `--allow-net` permission for any https package sources, even when running with `--cached-only` and not requiring actual network access. Is that intentional, or something that might be addressed in a future version? Security is a concern for my app, so I'm trying to be granular with permissions as much as possible.

**Comment by adamgreg** [comment]
Ah thank you! I did not know that option existed.
